repose-aggregator/artifacts/valve/src/main/scala/org/openrepose/valve/ReposeJettyServer.scala

```diff
@@ -126,37 +126,63 @@ class ReposeJettyServer(val nodeContext: AbstractApplicationContext,
 
         //Handle the Protocols and Ciphers
         //varargs are annoying, so lets deal with this using the scala methods
+        // From https://www.eclipse.org/jetty/documentation/9.4.x/configuring-ssl.html
+        // "When working with Includes / Excludes, it is important to know that Excludes will always win."
+        // TODO: The SslContextFactory's set Include/Exclude Protocols/CipherSuites override the built-in defaults.
+        // TODO: We may want to change at least the Excludes back to add's for v9.x.
         import scala.collection.JavaConversions._
-        Option(ssl.getExcludedProtocols).foreach(xp => cf.addExcludeProtocols(xp.getProtocol.toList: _*))
         Option(ssl.getIncludedProtocols).foreach { ip =>
           if (ip.getProtocol.nonEmpty) {
             cf.setIncludeProtocols(ip.getProtocol.toList: _*)
           }
-
         }
-        Option(ssl.getExcludedCiphers).foreach { xc =>
-          cf.addExcludeCipherSuites(xc.getCipher.toList: _*)
+        Option(ssl.getExcludedProtocols).foreach { ep =>
+          if (ep.getProtocol.nonEmpty) {
```

Comment:
So to sum up this comment and my others, I think we should:

1. Always set the includes/excludes (not add).
    1. This should be backwards compatible, and thus, just a small implementation change.
1. Allow empty sequences for includes/excludes (that the user has to explicitly declare to clear the defaults).
    1. This should be a new feature, since previously the user was not allowed to declare an empty sequence.

Reply:
1. Updated.
1. Current behavior on `included`'s is to overwrite (`set`), but current behavior on `excluded`'s is to append (`add`) to any defaults. This is what actually caused the issue with the `ReposeJettySSLTest` and led me down this path. So for backwards compatibility, `excluded`'s must remain `add`'s.
1. Any user defined `excluded-ciphers` are still being added. The difference here is to override the new Jetty defaults with the old ones. That is why the `TODO`'s to change these behaviors in Repose v9.x.

Reply:
In response to 2, I think your issue was with the defaults being updated. If you wipe the defaults and then set excludes, you should be okay. And from what I can tell, you only really need to wipe the cipher exclusion defaults, but I would wipe them all for consistency (both with Jetty 9.2.0 and with each other).

Reply:
Restructured.